python/lib/client/dmod/client/request_clients.py

```diff
@@ -351,6 +354,58 @@ def uses_auth(self) -> bool:
         pass
 
 
+class HttpDataTransferAgent(DataTransferAgent):
+    def __init__(self, http_client: aiohttp.ClientSession, ssl_context: Optional[ssl.SSLContext] = None):
+        self.http_client = http_client
+        self.ssl_context = ssl_context
+
+    async def download_dataset_item(self, dataset_name: str, item_name: str, dest: Path):
+        raise NotImplemented
+
+    async def upload_dataset_item(
+        self, dataset_name: str, item_name: str, source: Path
+    ) -> DatasetManagementResponse:
+        if not source.is_file():
+            return DatasetManagementResponse(
+                success=False,
+                reason="Dataset Upload File Not Found",
+                message=f"File {source!s} does not exist",
+            )
+
+        content_type, _ = mimetypes.guess_type(source)
+        form_data = aiohttp.FormData()
+        with source.open() as fp:
+            form_data.add_field(
+                "obj",
+                fp,
+                filename=item_name,
+                content_type=content_type,
+            )
+            response = await self.http_client.post(
+            "/add_object",
+            data=form_data,
+            params={
+                "dataset_name": dataset_name,
+                "object_name": item_name,
+            },
+            ssl=self.ssl_context,
+        )
+        if not response.ok:
+            return DatasetManagementResponse(
+                success=False,
+                reason="Dataset Item Upload Failed",
+                message=f"Status Code: {response.status}",
+            )
+
+        return DatasetManagementResponse(success=True, message="Dataset Item Upload Successful", reason="Success")
+
+    @property
+    def uses_auth(self) -> bool:
+        # NOTE: At this point in time, the dataset service's rest endpoints do
+        # not support auth
+        return False
```

Comment:
Let's make sure we have an issue on the project board tracking this.

Reply:
https://github.com/NOAA-OWP/DMOD/issues/648